river/tree/_nodes/base.py

```diff
@@ -80,25 +82,6 @@ def path(self, x) -> Iterator["Node"]:
     def iter_edges(self):
         yield None, 0, None, self, None
 
-    @property
```

Comment:
I see your point. However, as we don't make copies of the internal structures when returning the properties, one could still do something like:

```python
n_stats = node.stats  # property access

n_stats['class_name'] = 100000

print(node.stats['class_name'])
>> 100000
```
and the node's statistics would still be changed, right? :P

But nodes are a private module that is only accessed by the trees internally. So, it is (probably) fine to keep the nodes' attributes public. What do you think?

Reply:
Agree. Nodes are internal classes and it is nice to avoid the overhead from the properties. On the other hand, we must be careful to protect the nodes when exposing results to the user at the estimator level.